ibm/data_source_ibm_tg_gateways.go

```diff
@@ -0,0 +1,116 @@
+package ibm
+
+import (
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.ibm.com/ibmcloud/networking-go-sdk/transitgatewayapisv1"
+)
+
+func dataSourceIBMTransitGateways() *schema.Resource {
+	log.Println("inside of transit gateways:")
+	return &schema.Resource{
+		Read: dataSourceIBMTransitGatewaysRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
```

Comment:
Remove this "id" ...don't declate terraform will take care of id d.SetId(..)

Reply:
removed.